### PR TITLE
Add Sunlogin X v10.2.0.28522

### PR DESCRIPTION
**Pull request by ghost**
**Important:** *Do not tick a checkbox if you haven’t performed its action.* Honesty is indispensable for a smooth review process.

After making all changes to a cask, verify:

- [x] `brew cask audit --download {{cask_file}}` is error-free.
- [x] `brew cask style --fix {{cask_file}}` reports no offenses.
- [x] There are no [open pull requests](https://github.com/Homebrew/homebrew-cask/pulls) for the same update.
- [x] The submission is for [a stable version](https://github.com/Homebrew/homebrew-cask/blob/master/doc/development/adding_a_cask.md#stable-versions) or [documented exception](https://github.com/Homebrew/homebrew-cask/blob/master/doc/development/adding_a_cask.md#but-there-is-no-stable-version).

Additionally, **if adding a new cask**:

- [x] Named the cask according to the [token reference](https://github.com/Homebrew/homebrew-cask/blob/master/doc/cask_language_reference/token_reference.md).
- [x] `brew cask audit --new-cask {{cask_file}}` worked successfully.
- [x] `brew cask install {{cask_file}}` worked successfully.
- [x] `brew cask uninstall {{cask_file}}` worked successfully.
- [x] Checked the cask was not [already refused](https://github.com/Homebrew/homebrew-cask/search?q=is%3Aclosed&type=Issues).
- [x] Checked the cask is submitted to [the correct repo](https://github.com/Homebrew/homebrew-cask/blob/master/doc/development/adding_a_cask.md#finding-a-home-for-your-cask).

### CODE REVIEW

**Comment by miccal** [comment]
Please add `launchctl:` to the `uninstall` stanza -- from the CI:
```
Error: Some launch jobs were not unloaded, add them to uninstall launchctl:
com.oray.sunlogin.desktopagent
com.oray.sunlogin.helper
```

**Comment by miccal** [comment]
And one more according to the CI:
```
Error: Some launch jobs were not unloaded, add them to uninstall launchctl:
com.oray.sunlogin.service
```

File: Casks/sunloginclient.rb [hunk 1]
```diff
@@ -0,0 +1,19 @@
+cask 'sunloginclient' do
```
Comment (vitorgalvao) [inline Casks/sunloginclient.rb:1]: It’s the name of the app inside the pkg.

Comment (chocau) [reply Casks/sunloginclient.rb:1]: I think `sunlogin-x` is better.
1) There is already an app named `sunlogincontrol` in brew-cask repo. The corp calls the new app 'Sunlogin X for macOS' on its official website. Neither 'sunlogin' nor 'sunloginclient' is good.
2) This app does the same thing as Microsoft Remote Desktop, either remoter or remotee, and the word 'client' may be ambiguous.

Comment (vitorgalvao) [reply Casks/sunloginclient.rb:1]: > Neither 'sunlogin' nor 'sunloginclient' is good.

No one suggested `sunlogin`, and `sunloginclient` is the app’s name.

> This app does the same thing as Microsoft Remote Desktop

Irrelevant.

> and the word 'client' may be ambiguous.

Vendors’ bad app naming is generally not our concern. You may find the name ambiguous in relation to what the app does, but so far there’s no indication it’s ambiguous in relation to other apps. [It’s the user’s responsibility to know what the software is](https://github.com/homebrew/homebrew-cask/blob/master/doc/faq/not_a_discoverability_service.md) and the [token reference](https://github.com/caskroom/homebrew-cask/blob/master/doc/cask_language_reference/token_reference.md) is clear.

Everything in this points to `sinlogincleint`: the URL, the pkg, the installed app, and even the uninstallation. The only place where this is called `X` is the website, and not even the on the homepage address. We’ll allow the `X` in the `name` as it aids in search, but I see no reason why this cask would warrant a token exception.

**Pull request merged**